caffe2/utils/TypeTraits.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <caffe2/utils/C++17.h>
```

Comment:
btw, also on styleguide - do we use "" or <> include paths?

Reply:
Re relative path, the guide says

"""
Includes should use the full path (relative to fbcode/). The only exception is when the file being included resides in the same directory as the file including it -- then you can use the shorter #include "foo.h".
"""

since it is "can" rather than "should", it is probably easier to use "" and absolute path everywhere (see e.g. Folly https://github.com/facebook/folly/blob/master/folly/Memory.h#L30-L37. Folly uses <> though, but since style guide is using "" let's stick to that.